packages/webapp/src/_app/eos/secret-publisher.ts

```diff
@@ -0,0 +1,171 @@
+const ec = require("elliptic").ec("secp256k1");
+const bn = require("bn.js");
+
+export const publishSecretToChain = async (
+    message: string,
+    publisherAccount: string,
+    recipientAccounts: string[],
+    info?: string
+) => {
+    const [publisherKey, ...recipientKeys] = await fetchRecipientKeys([
+        publisherAccount,
+        ...recipientAccounts,
+    ]);
+    console.info(publisherKey, recipientKeys);
+
+    // TODO: check publisherkey matches the current one in browser
+
+    const transientKeyPair = await generateRandomTransientKey();
+    console.info(transientKeyPair);
+
+    const keks = await ecdhRecipientsKeyEncriptionKeys(
+        [publisherKey, ...recipientKeys],
+        transientKeyPair.privateKey,
+        info
+    );
+    console.info(keks);
+
+    const sessionKey = await generateRandomSessionKey();
+    console.info(sessionKey);
+
+    const encryptedSessionKeys = await encryptSessionKeys(sessionKey, keks);
+    console.info(encryptedSessionKeys);
+
+    const encryptedMessage = await encryptMessage(sessionKey, message);
+
+    // TODO: prepare publish trx to eos chain with all the prepared material
+    // eden.publishelection(
+    //     publisher = logged_in_user
+    //     round_no = '2',
+    //     encrypted_link,
+    //   // for each recipient
+    //     [(encrypted_session_keyN, recipient_keyN, transient_ecc_pubkey)]
+    //   )
+    console.info({
+        encryptedSessionKeys,
+        recipientKeys,
+        transientKeyPair,
+        encryptedMessage,
+    });
+
+    return {};
+};
+
+const fetchRecipientKeys = async (accounts: string[]) => {
+    // TODO: real implementation
+    // fetch eden.keys_table( where account = each account item )
+
+    const mockedKeysLen = accounts.length;
+    const mockedKeys = [];
+    for (let i = 0; i < mockedKeysLen; i++) {
+        mockedKeys.push(
+            "EOS5wGY8RcG8PuGYTPU9QwmiEm9XReBkK4SGtSuS4f1oeX4TNGaHT"
+        );
+        // temp PK: 5KcMypBPGByXbeBphsVzyyzUvg31tWNKsgrfXs2MmanYgYf4gao
+    }
+    return mockedKeys;
+};
+
+const generateRandomTransientKey = async () => {
+    const privateKey = await ecc.randomKey();
+    const publicKey = ecc.privateToPublic(privateKey);
+    return { privateKey, publicKey };
+};
+
+const ecdhRecipientsKeyEncriptionKeys = async (
+    recipientPublicKeys: string[],
+    transientPrivateKeyString: string,
+    info?: string
+) => {
+    const transientPrivateKey = PrivateKey.fromString(
+        transientPrivateKeyString
+    );
+    const transientEc = ec.keyPair({
+        priv: new bn(transientPrivateKey.d.toHex(), 16),
+    });
+    return await Promise.all(
+        recipientPublicKeys.map(async (publicKey: string) => {
+            const recipientEcPublicKey = ec.keyFromPublic(
+                PublicKey.fromString(publicKey).toHex(),
+                "hex"
+            );
+            const ecdhSecret = transientEc.derive(
+                recipientEcPublicKey.getPublic()
+            );
```

Comment:
@tbfleming I think it's a little nasty here since I'm using eosjs-ecc to handle eos keys but then converting it to `elliptic. secp256k1` format to derive the ecdh secret. Is there a more appropriated approach for doing this?

Reply:
eosjs-ecc is deprecated. eosjs replaced it with elliptic in 21.0.2. See the release notes for eosjs's conversion helpers: https://github.com/EOSIO/eosjs/releases/tag/v21.0.2

Reply:
`package.json`, `yarn.lock`, and `eosjs-ecc.d.ts` still pull in eosjs-ecc

Reply:
Nice catch! Removed!